Windows, CPR, lipstick fixes

phil235 opened the pull request:
- Fixes fulltile windows not spawning the correct amounts of shards and rods. Fixes #8139
- Fixes broken grille not dropping a rod when shot with buckshots.
- Fixes applying lipstick through masks. Fixes #8874
- Remove mob/proc/SpeciesCanConsume() , replaced by iscarbon checks.
- Remove alloweat var from masks, remove MASKCOVERSMOUTH flag from mime and clown masks so you can still eat through them.
- Some adjustments in surgery checks for bodyparts covered.
- Fixes a runtime in weapon/energy attack_verb.
- Humans can now CPR monkeys. Fixes #7310 
- Add feedback when doing CPR on a corpse. Fixes #8707
- Slight changes to CPR code.
- Fixes runtime with drones trying to do CPR. Fixes #5553 (it's the last runtime remaining)
- Rewrites /obj/item/weapon/reagent_containers/proc/canconsume to be less shitty.
- Adds an is_mouth_covered() proc to carbons to check for mask and headgear with MASKCOVERSMOUTH or HEADCOVERSMOUTH, with argument to restrict the check to only mask or only head if needed.

phil235 committed:
Fixes fulltile windows not spawning the correct amounts of shards and rods.
Fixes broken grille not dropping a rod when shot with buckshots.

WJohn commented:
I kinda wish fulltile windows would break into 2 shards when broken. If the point of moving all windows to fulltile was to make repair easier, then having it be made so you can't repair damage easily is really dumb.

phil235 commented:
The fulltile windows DO drop two shards with this PR.

WJohn commented:
Ah, very nice.

phil235 committed:
Remove SpeciesCanConsume() proc, replaced by iscarbon checks.

Remove alloweat var from masks, remove MASKCOVERSMOUTH flag from mime and clown masks so you can still eat through them.

Some adjustments in surgery checks for bodyparts covered.
Fixes a runtime in weapon/energy attack_verb.

Humans can now CPR monkeys.
Slight changes to CPR code.
Fixes runtime with drones trying to do CPR.

Rewrites /obj/item/weapon/reagent_containers/proc/canconsume to be less shitty.

Adds an is_mouth_covered() proc to carbons to check for mask and headgear with MASKCOVERSMOUTH or HEADCOVERSMOUTH, with argument to restrict the check to only mask or only head.
